lib/util/git/modified-cache.ts

```diff
@@ -24,12 +24,12 @@ export function setCachedModifiedResult(
   const cache = getCache();
   cache.branches ??= [];
   const { branches } = cache;
-  const branch =
-    branches?.find((branch) => branch.branchName === branchName) ??
-    ({ branchName: branchName } as BranchCache);
+  let branch = branches?.find((branch) => branch.branchName === branchName);
 
   // if branch not present add it to cache
-  if (is.undefined(branch.sha)) {
+  if (!branch) {
+    logger.warn(`Branch cache not present for ${branchName}`);
```

Comment:
This should never happen?

Reply:
Yes now it won't wasn't sure whether to warn or use thee old code, need suggestion

Reply:
In fact, I was able to locate an instance of this. It occurs while eliminating stale branches, thus it's a good idea not to create new branch caches here as doing so will cause us to store an extra branch during the cleanup cycle. 

How to reproduce:
1. Create a standard repository with dependencies set between 0.0.0 the most recent version.
2. Run renovate -> update PR created
3. Adjust the version number in the base branch to greater most recent version. (if most recent is 1.2.0 update it to 1.2.3)
4. Run renovate once more 

Result:
![image](https://user-images.githubusercontent.com/76607510/190527084-f8924f4c-4102-4e79-87c8-27325b44fae0.png)

Reply:
I don't think we should warn then, if it's possible normal behavior